src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -644,11 +645,17 @@ def todict(obj, post_processor=None):
         return obj.isoformat()
     if isinstance(obj, timedelta):
         return str(obj)
-    # This is the only difference with knack.util.todict because for typespec generated SDKs
-    # The base model stores data in obj.__dict__['_data'] instead of in obj.__dict__
-    # We need to call obj.as_dict() to extract data for this kind of model
-    if hasattr(obj, 'as_dict') and not hasattr(obj, '_attribute_map'):
-        result = {to_camel_case(k): todict(v, post_processor) for k, v in obj.as_dict().items()}
+    # This is the only difference with knack.util.todict
+    # azure-core has provided
+    #   - `is_generated_model` to identify auto generated SDK models
+    #   - `attribute_list` to list all attribute names
+    # no matter it's track1/track2/swagger generated/typespec generated SDKs
+    # no matter what transformation (renaming/formatting/...) the attribute has during auto generation
+    if is_generated_model(obj):
+        result = {}
+        for attr in attribute_list(obj):
+            if hasattr(obj, attr):
```

Comment:
We need to check whether the attr exist because we may import SDK model and then do some customization like keyvault certificate L122-127 https://github.com/Azure/azure-cli/blob/bb2b31a832033e1b6e1343cc49b8ec023cd537fb/src/azure-cli/azure/cli/command_modules/keyvault/custom.py#L69-L128

Some attributes are deleted